test/parallel/test-runner-watch-mode.mjs

```diff
@@ -0,0 +1,52 @@
+import '../common/index.mjs';
+import { describe, it } from 'node:test';
+import { spawn } from 'node:child_process';
+import { writeFileSync } from 'node:fs';
+import * as fixtures from '../common/fixtures.mjs';
+
+function deferred() {
```

Comment:
Optional, but we have `createDeferredPromise()` in `internal/util`. The only catch is that the test would require `--expose-internals`.